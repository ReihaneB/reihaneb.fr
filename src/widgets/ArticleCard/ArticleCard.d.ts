import { ImageProps } from '@/components/Image/Image';

export interface ArticleCardProps {
  id: number;
  title: string;
  image: ImageProps;
  to: string;
  description: string;
  readTime: number;
  color: string;
  category: {
    title: string;
    color: string;
  };
}
