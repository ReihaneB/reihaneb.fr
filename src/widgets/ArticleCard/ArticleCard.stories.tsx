import type { Meta, StoryObj } from '@storybook/react';
import { expect, userEvent, waitFor, within } from '@storybook/test';

import { lokiPlayDecorator } from '@/testUtils/loki/play-decorator';
import _16x9 from '@/styles/assets/placeholder/_16x9.png';
import _4x3 from '@/styles/assets/placeholder/_4x3.png';
import _1x1 from '@/styles/assets/placeholder/_1x1.png';

import ArticleCard from './ArticleCard';

import styles from './ArticleCard.stories.module.css';

const meta: Meta<typeof ArticleCard> = {
  title: 'Widgets/ArticleCard',
  component: ArticleCard,
  decorators: [
    Story => (
      <div className={styles.container}>
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof ArticleCard>;

const defaultArgs = {
  id: 1,
  title: 'Id consectetur adipisicing Lorem aute proident.',
  image: {
    src: _1x1,
    alt: 'Article image',
    width: 400,
    height: 400,
  },
  color: '#865959',
  category: {
    title: 'CSS',
    color: '#ffdede',
  },
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.',
  readTime: 5,
  to: 'article-slugged',
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const WithLongTitle: Story = {
  args: {
    ...defaultArgs,
    title: 'Quis proident deserunt reprehenderit non consectetur cillum est velit eiusmod deserunt id cillum non.',
  },
};

export const WithLongDescription: Story = {
  args: {
    ...defaultArgs,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.',
  },
};

export const _1x1Image: Story = {
  args: {
    ...defaultArgs,
  },
};

export const _4x3Image: Story = {
  args: {
    ...defaultArgs,
    image: {
      src: _4x3,
      alt: 'Article image',
      width: 400,
      height: 300,
    },
  },
};

export const _16x9Image: Story = {
  args: {
    ...defaultArgs,
    image: {
      src: _16x9,
      alt: 'Article image',
      width: 400,
      height: 225,
    },
  },
};

export const LongCategoryName: Story = {
  args: {
    ...defaultArgs,
    category: {
      title: 'This is a very long category name',
      color: '#ffdede',
    },
  },
};

export const LongReadTime: Story = {
  args: {
    ...defaultArgs,
    readTime: 100,
  },
};

export const OnlyFocusLink: Story = {
  args: {
    ...defaultArgs,
  },
  play: lokiPlayDecorator(
    async ({ canvasElement }) => {
      const canvas = within(canvasElement);

      await userEvent.tab();

      await waitFor(() => expect(canvas.getByTestId('link-component')).toHaveFocus());
    }
  ),
};

export const LinkHaveTheRightHref: Story = {
  args: {
    ...defaultArgs,
    to: 'article-slugged',
  },
  play: lokiPlayDecorator(
    async ({ canvasElement }) => {
      const canvas = within(canvasElement);

      await waitFor(() => expect(canvas.getByTestId('link-component')).toHaveAttribute('href', '/blog/article-slugged'));
    }
  ),
};
