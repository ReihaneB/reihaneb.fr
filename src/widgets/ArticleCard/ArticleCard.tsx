import React from 'react';

import Image from '@/components/Image/Image';
import Link from '@/components/Link/Link';
import Tag from '@/components/Tag/Tag';
import routes from '@/utils/routes';

import styles from './ArticleCard.module.css';
import type { ArticleCardProps } from './ArticleCard.d';


function ArticleCard({
  id,
  title,
  description,
  readTime,
  category,
  color,
  image,
  to,
  ...rest
}: ArticleCardProps) {
  return (
    <article
      className={styles.root}
      style={{ '--article-card-color': color } as React.CSSProperties}
      {...rest}
    >
      <div className={styles.imageContainer}>
        <Image
          src={image.src}
          alt={image.alt}
          width={image.width}
          height={image.height}
          priority
        />
      </div>
      <div className={styles.content}>
        <div className={styles.tags}>
          <Tag
            color={category.color}
          >
            {category.title}
          </Tag>
          <Tag>
            {readTime}
            {' '}
            min.
          </Tag>
        </div>
        <h2 className={styles.title}>
          {title}
        </h2>
        <p className={styles.description}>
          {description}
        </p>
        <Link
          to={routes.ARTICLE_WITH_ID(to)}
          fullWidth
        >
          Continuer la lecture
        </Link>
      </div>
    </article>
  );
}

export default ArticleCard;
