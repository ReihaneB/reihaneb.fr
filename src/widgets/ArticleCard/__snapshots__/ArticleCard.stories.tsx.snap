// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/ArticleCard _1x1Image smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="400"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
             src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard _4x3Image smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="300"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_4x3.58bee57f.png?w=640&amp;q=75 1x, /static/media/_4x3.58bee57f.png?w=828&amp;q=75 2x"
             src="/static/media/_4x3.58bee57f.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard _16x9Image smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="225"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_16x9.d50e36d3.png?w=640&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=828&amp;q=75 2x"
             src="/static/media/_16x9.d50e36d3.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard LinkHaveTheRightHref play-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="400"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
             src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard LongCategoryName smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="400"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
             src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            This is a very long category name
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard LongReadTime smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="400"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
             src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            100 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard OnlyFocusLink play-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="400"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
             src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="400"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
             src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard WithLongDescription smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="400"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
             src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Id consectetur adipisicing Lorem aute proident.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;

exports[`Widgets/ArticleCard WithLongTitle smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-ArticleCard-ArticleCard-stories-module__container--Pe2pq">
    <article class="src-widgets-ArticleCard-ArticleCard-module__root--avFaE"
             style="--article-card-color: #865959;"
    >
      <div class="src-widgets-ArticleCard-ArticleCard-module__imageContainer--jGYSb">
        <img alt="Article image"
             fetchpriority="high"
             width="400"
             height="400"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
             src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
             style="color: transparent;"
        >
      </div>
      <div class="src-widgets-ArticleCard-ArticleCard-module__content--FrurJ">
        <div class="src-widgets-ArticleCard-ArticleCard-module__tags--IqAYc">
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
               style="--tag-color: #ffdede;"
          >
            CSS
          </div>
          <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
               data-testid="tag-component"
               tabindex="-1"
          >
            5 min.
          </div>
        </div>
        <h2 class="src-widgets-ArticleCard-ArticleCard-module__title--dWwUy">
          Quis proident deserunt reprehenderit non consectetur cillum est velit eiusmod deserunt id cillum non.
        </h2>
        <p class="src-widgets-ArticleCard-ArticleCard-module__description--IjJJQ">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vehicula, turpis vel luctus tincidunt, erat felis lacinia libero, nec accumsan nunc odio sit amet urna. Sed at urna auctor, tincidunt lacus vitae, ultricies nulla. Nullam euismod, justo sit amet tincidunt vehicula, nunc ex ultricies felis, nec lacinia nunc dolor vel libero. Nulla facilisi. Nullam eget nunc ac libero ultricies ultricies. Suspendisse potenti. Aliquam erat volutpat. Nullam nec ante auctor, ultricies elit ac, lacinia libero.
        </p>
        <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__fullWidth--Rafql"
           data-testid="link-component"
           target="_self"
           href="/blog/article-slugged"
        >
          Continuer la lecture
        </a>
      </div>
    </article>
  </div>
</main>
`;
